streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -349,6 +360,119 @@ public String queryableStoreName() {
         return toStream().selectKey(mapper);
     }
 
+    @Override
+    public KTable<K, V> suppress(final Suppress<K, V> suppress) {
+        final SuppressImpl<K, V> suppressImpl = (SuppressImpl<K, V>) suppress;
+        final String name = builder.newProcessorName(SUPPRESS_NAME);
+
+        final ProcessorSupplier<K, Change<V>> suppressionSupplier =
+            () -> new KTableSuppressProcessor<>(buildSuppress(suppressImpl));
+
+        final ProcessorParameters<K, Change<V>> processorParameters = new ProcessorParameters<>(
+            suppressionSupplier,
+            name
+        );
+
+        final ProcessorGraphNode<K, Change<V>> node = new ProcessorGraphNode<>(name, processorParameters, false);
+
+        builder.addGraphNode(this.streamsGraphNode, node);
+
+        return new KTableImpl<K, S, V>(
+            builder,
+            name,
+            suppressionSupplier,
+            keySerde,
+            valSerde,
+            Collections.singleton(this.name),
+            null,
+            false,
+            node
+        );
+    }
+
+    @SuppressWarnings("unchecked") // these casts are silly
+    private SuppressImpl<K, V> buildSuppress(final SuppressImpl<K, V> suppress) {
+        if (suppress.isFinalResultsSuppression()) {
+            // This throws an exception if the computation is not windowed
+            final long grace = findAndVerifyWindowGrace(streamsGraphNode, "");
+
+            // so we know we can cast the key to Windowed
+            final SuppressImpl<? extends Windowed, V> windowedSuppress = (SuppressImpl<? extends Windowed, V>) suppress;
+
+            final Suppress<? extends Windowed, V> finalResultsSuppression = SuppressImpl.buildFinalResultsSuppression(
+                Duration.ofMillis(grace),
+                windowedSuppress
+            );
+
+            //
+            return (SuppressImpl<K, V>) finalResultsSuppression;
+        } else {
+            return suppress;
+        }
+    }
+
+    private static long findAndVerifyWindowGrace(final StreamsGraphNode streamsGraphNode, final String chain) {
+        // error base case: we traverses off the end of the graph without finding a window definition
+        if (streamsGraphNode == null) {
+            throw new IllegalArgumentException(
```

Comment:
Let's use `TopologyException` instead of `IllegalArgumentException` here, ditto below.

Reply:
Ok. Thanks. That sounds much better.